test-suite/testkit/tests/api/main.rs

```diff
@@ -132,3 +134,35 @@ fn moved() {
         .expect_err("Request to the `MovedPermanently` endpoint succeed");
     assert_matches!(pong_error, ApiError::MovedPermanently(_));
 }
+
+/// Checks response from endpoint with new error type.
+#[test]
+fn endpoint_with_new_error_type() {
+    let (_testkit, api) = init_testkit();
+
+    // Check OK response.
+    let ok_query = PingQuery { value: 64 };
+    let response: u64 = api
+        .public(ApiKind::Service("api-service"))
+        .query(&ok_query)
+        .get_new::<u64>("new-error-type")
```

Comment:
Nit: Turbofish is redundant here since you've specified `response` type earlier.